### PR TITLE
feat: Option to automatically turn snake_case columns in a query response into camelCase fields #310

### PR DESCRIPTION
**Pull request by pantapita**
I've been using this feature and it works well (see the proposed changes). I didn't implement the switch as it should be compliant with the vision of the authors, so please advise where to put the switch and how to name it ?
I'm addressing "query.ts -> snakeToCamel" in particular !

Best

### CODE REVIEW

**Comment by Soremwar** [comment]
Hi @pantapita, the option to rename the fields to camelcase should be added here
https://github.com/denodrivers/postgres/blob/235d57568bd3a49c353a988918d06077197e9044/query/query.ts#L53
in a per query basis

Thanks for your contribution!

File: query/query.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export interface QueryConfig {
   encoder?: (arg: unknown) => EncodedArg;
   name?: string;
   text: string;
+  snakeToCamel?: boolean;
```
Comment (Soremwar) [inline query/query.ts:52]: Name the argument as camelcase, i'll take care of adding it to the docs later

Comment (Soremwar) [inline query/query.ts:52]: Add the option to the QueryObjectConfig interface, since QueryConfig applies to queryArray which contains no query names

**Review by Soremwar — CHANGES_REQUESTED** [review]

**Comment by Soremwar** [comment]
While reviewing I realized that this implementation only renames the fields passed to the `fields` argument, instead of renaming the fields returned naturally by the database, which is the feature requested by the issue #310

I don't think it's worth adding to the codebase unless that particular issue is addressed, since the `fields` parameter can be all camelcase by simple action of the user

**Comment by lucacasonato** [comment]
If it helps, this is the conversion function I used for a local project of mine:

```ts
/** Given an object that uses snake_case keys, turn it into an object using camelCase keys */
export function camelCaseify<T>(obj: Record<string, unknown>): T {
  const result: Record<string, unknown> = {};
  for (const snakeCasedKey in obj) {
    const camelCasedKey = snakeCasedKey
      .replaceAll(/_([a-z])/g, (g) => g[1].toUpperCase());
    result[camelCasedKey] = obj[snakeCasedKey];
  }
  return result as T;
}
```

**Comment by Soremwar** [comment]
Superseded by #333

**Pull request closed without merging**